shard-operator-parent/shard-operator-v2/src/main/java/com/redhat/service/smartevents/shard/operator/v2/resources/CamelIntegrationStatus.java

```diff
@@ -1,48 +1,31 @@
 package com.redhat.service.smartevents.shard.operator.v2.resources;
 
-import java.util.Arrays;
 import java.util.HashSet;
-import java.util.Optional;
 import java.util.Set;
 
-import com.fasterxml.jackson.annotation.JsonIgnore;
 import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
 import com.redhat.service.smartevents.shard.operator.core.resources.Condition;
 import com.redhat.service.smartevents.shard.operator.core.resources.ConditionStatus;
 
 @JsonIgnoreProperties(ignoreUnknown = true)
-public class CamelIntegrationStatus {
+public class CamelIntegrationStatus extends BaseResourceStatus {
```

Comment:
Changed to extend sub-class to avoid duplication of `isReady()` logic and `getConditionByType()`